### PR TITLE
Expose `actix::ContextFut::restart` to public API

### PR DESCRIPTION
**Pull request by 50U10FCA7**
## PR Type

Feature

## PR Checklist

Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

### Motivation

Making possible to implement custom [`actix::Supervisor`] types.

### Problem

[`actix::Supervisor`] does not allow to shutdown an [`actix::Actor`] it supervising. In this case a custom [`actix::Supervisor`] should be implemented by the user to satisfy it needs (graceful shutdown of a supervised [`actix::Actor`] via `Supervisor::shutdown` in my case for example). But there is no way to implement a custom [`actix::Supervisor`] type because [`actix::ContextFut::restart`] is crate-local and can't be used in usercode.

### Solution

Make [`actix::ContextFut::restart`] public.

[`actix::Actor`]: https://github.com/actix/actix/blob/d7aab0a0f439679a570d08aaa6972ad8b60003de/actix/src/actor.rs#L75
[`actix::ContextFut::restart`]: https://github.com/actix/actix/blob/d7aab0a0f439679a570d08aaa6972ad8b60003de/actix/src/contextimpl.rs#L281
[`actix::Supervisor`]: https://github.com/actix/actix/blob/d7aab0a0f439679a570d08aaa6972ad8b60003de/actix/src/supervisor.rs#L74

### CODE REVIEW

File: actix/CHANGES.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 - Add `SyncArbiter::start_with_thread_builder()`.
 - Derive `PartialEq` and `Eq` for `MailboxError`.
+- Expose `ContextFut::restart`.
```
Comment (robjtede) [inline actix/CHANGES.md:11]: move to "unreleased"

File: actix/src/contextimpl.rs [hunk 2]
```diff
@@ -278,7 +278,7 @@ where
 
     /// Restart context. Cleanup all futures, except address queue.
     #[inline]
-    pub(crate) fn restart(&mut self) -> bool
+    pub fn restart(&mut self) -> bool
```
Comment (robjtede) [inline actix/src/contextimpl.rs:281]: if we're making this public, i'd like to see a test added somewhere and also some improved documentation on this method

**Comment by 50U10FCA7** [comment]
@robjtede Can you please check the changes?

**Comment by 50U10FCA7** [comment]
@robjtede Any updates on this PR?

**Comment by 50U10FCA7** [comment]
Merged the last changes from `master`.

**Comment by 50U10FCA7** [comment]
@robjtede Can you review this PR please?

**Comment by 50U10FCA7** [comment]
@robjtede ping

**Review by robjtede — APPROVED** [review]
lgtm, apologies for the delay 👍🏻

**Pull request merged**